### ISSUE TITLE
Notes sync automatically regardless of settings

### ISSUE THREAD

**Issue by acarasimon96** [laurent22/joplin#2191]
Joplin seems to sync my notes too frequently, i.e. every minute after every time I do something with them. When I set the sync interval from 5 minutes to Disabled, it still automatically syncs. This behavior still occurs when the sync interval is set to longer than 5 minutes. Not even restarting the app after changing the settings have no effect on this.

On Windows, I could tell when this does happen by seeing my custom editor font suddenly becoming blurry, obviously because Chrome's font rendering sucks in that OS.

## Environment

- Joplin version: 1.0.175
- Platform: Windows, Linux
- OS specifics: Windows 10 Pro 1803, Manjaro (installed via AUR)

## Steps To Reproduce

1. Make sure sync is set up (I used OneDrive)
2. Make any changes (e.g. creating new note, editing an existing note)
3. Wait about a minute
4. Observe the Synchronize button on the lower left corner change
5. Change the sync interval setting and repeat steps 2-3

## Expected behavior

The app would sync *only* after a specified amount of time as set in the Synchronization settings or never when set to Disabled.

## Logfiles

- [log_20191215.txt](https://github.com/laurent22/joplin/files/3965203/log_20191215.txt)
- [log-database_20191215.txt](https://github.com/laurent22/joplin/files/3965205/log-database_20191215.txt)

**Comment by laurent22** [comment]
Not a bug, but UI needs to be clarified

**Comment by acarasimon96** [comment]
So are you saying that the behavior described above is a feature? I still believe it looks more of like a bug to me.

**Comment by tessus** [comment]
This has been discussed a few times on the forum.

e.g. https://discourse.joplinapp.org/t/config-sync-interval-0-not-respected/4641/2?u=tessus

**Comment by acarasimon96** [comment]
@tessus I forgot it was already brought up in the forums, but that makes a lot of sense now. Thanks

**Comment by Perkolator** [comment]
> This has been discussed a few times on the forum.
> 
> e.g. https://discourse.joplinapp.org/t/config-sync-interval-0-not-respected/4641/2?u=tessus

From the forum:

> There are 2 sync operations:
> 
> 1. a short update to the sync target when a local file changes (you can’t turn this off)
> 2. a sync operation that retrieves changes from the sync target (the sync interval)

Why is the 1st mandatory/forced?

The "upload-sync after 30 seconds of inactivity of an edited note" feature might have a part on the reason why some characters/text is sometimes lost in notes. I wrote this in another issue:

> There have been reports about some last character(s) missing sometimes after a sync. There might be a chance that the desktop "30 sec auto-save + auto upload sync" is the culprit. I haven't been able to reproduce the problem consistently yet. I believe it has something to do with timing, i.e. when the auto-save/auto-upload-sync starts or during its procedure and editing the note at the same time. I don't know if I have understood the system correctly, but I think that when the auto-save/auto-upload-sync starts or during it, Joplin makes a "snapshot" of the note text and upload syncs that, but in the same time user has added something to the note and the note view is not updated after the upload-sync. Only after "regular/full" sync happens (upload+download), then the note text is fetched that Joplin only upload-synced, and the note view is updated, and the late modification that happened during the auto-save/auto-sync is removed from the view. I'm not sure about this but I think that this hypothesis would be worth checking out.

Also why does the mobile have to have a similar forced auto-save/auto-upload-sync that triggers after 1 second (might be even less actually, like 500ms) of inactivity?
